gnovm/pkg/gnolang/store.go

```diff
@@ -775,8 +826,8 @@ func (ds *defaultStore) LogSwitchRealm(rlmpath string) {
 
 func (ds *defaultStore) ClearCache() {
```

Comment:
do you think that we should protect these map modifications behind a lock?

Reply:
Well, no, but actually, yes (but it's out of scope IMO).

As you may have noticed, there are no locks in this Store implementation. This is because it's generally not used concurrently; VM transactions are executed sequentially, and queries run on transactions that are never committed. When it comes to ClearCache, specifically, this function is only executed for filetests and when explicitly called in testing; ie. never on-chain, so it's not so much of a problem IMO.

However... I do think there's race conditions here that we should address. Like: what happens when transactions are being executed at the same time of a query? The query could access the maps while the gno store is partially modified; or do a concurrent map read+write (which is a panic). So I think there is probably some locking to be done (unless it happens at the tm2 level). But I think it's reasonable for the locking to not pertain to Gno's store specifically; rather, to have the gno store as one of the mutex'ed values.